ci/scripts/java_jni_build.sh

```diff
@@ -56,7 +68,7 @@ cmake \
   -DBUILD_TESTING=${ARROW_JAVA_BUILD_TESTS} \
   -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
   -DCMAKE_PREFIX_PATH=${arrow_install_dir} \
-  -DCMAKE_INSTALL_LIBDIR=lib \
+  -DCMAKE_ARCH_DIR=${normalized_arch} \
```

Comment:
Please don't use `CMAKE_` prefix for our variable. Please use `ARROW_JAVA_JNI_` prefix like other our variables such as `ARROW_JAVA_JNI_ENABLE_DEFAULT` instead.

BTW, why do we need to install to sub directory such as `lib/arrow_cdata_jni/${arch}/libarrow_cdata_jni.so` instead of `lib/${arch}/libarrow_cdata_jni.so`? Nameing conflict? We don't have distinct base name such `libarrow_cdata_jni` and `libarrow_dataset_jni`. So I think that no naming conflict is happen.

Reply:
Sure, I'll fix the name of the ARCH_DIR variable.

Yes the reason is a naming conflict. It's only visible when using JPMS modules.
It's because we see multiple JPMS modules (where a JPMS module is a JAR with a module-info.class file) that contain the same subfolder at the root level (the ${arch} folder). JPMS sees this as a package collision even though the directory only contains binaries and throws startup exceptions in this scenario.